src/main/java/ru/yandex/clickhouse/jdbcbridge/db/clickhouse/ClickHouseConverter.java

```diff
@@ -33,46 +18,100 @@
+        map.put(new TypeWithScaleAndPrecision(FLOAT,null,null), new MappingInstruction<>(Float32, ResultSet::getFloat, (i, s) -> s.writeFloat32(i)));
+        map.put(new TypeWithScaleAndPrecision(REAL,null,null), new MappingInstruction<>(Float32, ResultSet::getFloat, (i, s) -> s.writeFloat32(i)));
+        map.put(new TypeWithScaleAndPrecision(DOUBLE,null,null), new MappingInstruction<>(Float64, ResultSet::getFloat, (i, s) -> s.writeFloat64(i)));
 
-        map.put(TIMESTAMP, new MappingInstruction<>(DateTime, ResultSet::getTimestamp, (i, s) -> s.writeDateTime(i)));
-        map.put(TIME, new MappingInstruction<>(DateTime, ResultSet::getTime, (i, s) -> s.writeDateTime(i)));
-        map.put(DATE, new MappingInstruction<>(Date, ResultSet::getDate, (i, s) -> s.writeDate(i)));
+        map.put(new TypeWithScaleAndPrecision(TIMESTAMP,null,null), new MappingInstruction<>(DateTime, ResultSet::getTimestamp, (i, s) -> s.writeDateTime(i)));
+        map.put(new TypeWithScaleAndPrecision(TIME,null,null), new MappingInstruction<>(DateTime, ResultSet::getTime, (i, s) -> s.writeDateTime(i)));
+        map.put(new TypeWithScaleAndPrecision(DATE,null,null), new MappingInstruction<>(Date, ResultSet::getDate, (i, s) -> s.writeDate(i)));
 
-        map.put(BIT, new MappingInstruction<>(UInt8, ResultSet::getBoolean, (i, s) -> s.writeUInt8(i)));
-        map.put(BOOLEAN, new MappingInstruction<>(UInt8, ResultSet::getBoolean, (i, s) -> s.writeUInt8(i)));
+        map.put(new TypeWithScaleAndPrecision(BIT,null,null), new MappingInstruction<>(UInt8, ResultSet::getBoolean, (i, s) -> s.writeUInt8(i)));
+        map.put(new TypeWithScaleAndPrecision(BOOLEAN,null,null), new MappingInstruction<>(UInt8, ResultSet::getBoolean, (i, s) -> s.writeUInt8(i)));
 
         // @todo test
-        map.put(CHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
-        map.put(VARCHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
-        map.put(LONGVARCHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
-        map.put(NVARCHAR, new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(CHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(VARCHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(LONGVARCHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+        map.put(new TypeWithScaleAndPrecision(NVARCHAR,null,null), new MappingInstruction<>(String, ResultSet::getString, (i, s) -> s.writeString(i)));
+
+        for (int precision = 1 ; precision <= 38 ; precision++ ){
```

Comment:
Please, use constants for magic numbers.

Reply:
I added DECIMAL_XX_MAXPRECISION constants